### ISSUE TITLE
Contour start point compatibility issue with a couple of glyphs

### ISSUE THREAD

**Issue by behdad** [googlefonts/googlesans-flex#716]
Discovered using the `varLib.interpolatable` tool:
```
% ./fonttools varLib.interpolatable googlesans-flex-fonts/fonts/variable/GoogleSansFlex\[ROND,opsz,slnt,wdth,wght\].ttf                                                                               
Glyph .notdef was not compatible: 
    Contour 0 start point differs: (), (('opsz', -1.0), ('wdth', -0.66668701171875))
    Contour 1 start point differs: (), (('opsz', -1.0), ('wdth', -0.66668701171875))
    Contour 0 start point differs: (), (('opsz', -6.103515625e-05), ('wdth', -0.66668701171875))
    Contour 1 start point differs: (), (('opsz', -6.103515625e-05), ('wdth', -0.66668701171875))
    Contour 0 start point differs: (), (('opsz', -1.0), ('wdth', -0.66668701171875), ('wght', 1.0))
    Contour 1 start point differs: (), (('opsz', -1.0), ('wdth', -0.66668701171875), ('wght', 1.0))
Glyph acutecomb was not compatible: 
    Contour 0 start point differs: (), (('ROND', 1.0), ('opsz', 1.0), ('wdth', 1.0), ('wght', -1.0))
Glyph degree was not compatible: 
    Contour order differs: [0, 1] in (), [1, 0] in (('opsz', 1.0), ('wght', -1.0))
Glyph divide was not compatible: 
    Contour 0 start point differs: (), (('wdth', -1.0),)
    Contour 1 start point differs: (), (('wdth', -1.0),)
    Contour 0 start point differs: (), (('wdth', -0.66668701171875),)
    Contour 1 start point differs: (), (('wdth', -0.66668701171875),)
    Contour 0 start point differs: (), (('wdth', 1.0),)
    Contour 1 start point differs: (), (('wdth', 1.0),)
    Contour 0 start point differs: (), (('ROND', 1.0), ('wdth', -1.0))
    Contour 1 start point differs: (), (('ROND', 1.0), ('wdth', -1.0))
    Contour 0 start point differs: (), (('ROND', 1.0), ('wdth', -0.66668701171875))
    Contour 1 start point differs: (), (('ROND', 1.0), ('wdth', -0.66668701171875))
    Contour 0 start point differs: (), (('ROND', 1.0), ('wdth', 1.0))
    Contour 1 start point differs: (), (('ROND', 1.0), ('wdth', 1.0))
    Contour 0 start point differs: (), (('opsz', -6.103515625e-05), ('wdth', -1.0))
    Contour 1 start point differs: (), (('opsz', -6.103515625e-05), ('wdth', -1.0))
    Contour 0 start point differs: (), (('slnt', -1.0), ('wdth', -1.0))
    Contour 1 start point differs: (), (('slnt', -1.0), ('wdth', -1.0))
    Contour 0 start point differs: (), (('slnt', -1.0), ('wdth', -0.66668701171875))
    Contour 1 start point differs: (), (('slnt', -1.0), ('wdth', -0.66668701171875))
    Contour 0 start point differs: (), (('slnt', -1.0), ('wdth', 1.0))
    Contour 1 start point differs: (), (('slnt', -1.0), ('wdth', 1.0))
    Contour 0 start point differs: (), (('ROND', 1.0), ('slnt', -1.0), ('wdth', -1.0))
    Contour 1 start point differs: (), (('ROND', 1.0), ('slnt', -1.0), ('wdth', -1.0))
    Contour 0 start point differs: (), (('ROND', 1.0), ('slnt', -1.0), ('wdth', -0.66668701171875))
    Contour 1 start point differs: (), (('ROND', 1.0), ('slnt', -1.0), ('wdth', -0.66668701171875))
    Contour 0 start point differs: (), (('ROND', 1.0), ('slnt', -1.0), ('wdth', 1.0))
    Contour 1 start point differs: (), (('ROND', 1.0), ('slnt', -1.0), ('wdth', 1.0))
    Contour 0 start point differs: (), (('opsz', -6.103515625e-05), ('slnt', -1.0), ('wdth', -1.0))
    Contour 1 start point differs: (), (('opsz', -6.103515625e-05), ('slnt', -1.0), ('wdth', -1.0))
Glyph fraction was not compatible: 
    Contour 0 start point differs: (), (('opsz', 1.0), ('slnt', -1.0), ('wdth', -0.20001220703125), ('wght', -1.0))
    Contour 0 start point differs: (), (('ROND', 1.0), ('opsz', 1.0), ('slnt', -1.0), ('wdth', -0.20001220703125), ('wght', -1.0))
Glyph hungarumlaut was not compatible: 
    Contour 0 start point differs: (), (('ROND', 1.0), ('opsz', 1.0), ('wdth', 1.0), ('wght', -1.0))
Glyph uni030B was not compatible: 
    Contour 0 start point differs: (), (('ROND', 1.0), ('opsz', 1.0), ('wdth', 1.0), ('wght', -1.0))
```
I have manually verified that the `divide` glyph is a real issue. `.notdef` is also probably a real issue. I have not been able to reproduce the problem with `acutecomb` yet. Will report back if I can.

**Comment by behdad** [comment]
To reproduce the `divide` issue, we focus on these lines of the report:
```
Glyph divide was not compatible: 
    Contour 0 start point differs: (), (('wdth', -1.0),)
    Contour 1 start point differs: (), (('wdth', -1.0),)
```
It says that the contour start points for contours 0 & 1 are different between the default (`()`) and minimum-width (`(('wdth', -1.0),)`) masters.  Now, we visualize those and the in-between:

For default master:
```
$ ~/harfbuzz/build/util/hb-view googlesans-flex-fonts/fonts/variable/GoogleSansFlex\[ROND,opsz,slnt,wdth,wght\].ttf --glyphs divide > out.png
```
![out](https://github.com/googlefonts/googlesans-flex/assets/142891/a607f7c1-e679-4f90-b82c-5105caf0b01f)

For minimum-width master:
```
$ ~/harfbuzz/build/util/hb-view googlesans-flex-fonts/fonts/variable/GoogleSansFlex\[ROND,opsz,slnt,wdth,wght\].ttf --glyphs divide --variations wdth=25 > out.png
```
![out](https://github.com/googlefonts/googlesans-flex/assets/142891/32a64fa0-d429-4b41-8a07-ecc6219be41c)

Now let's look at the in-between:
```
$ ~/harfbuzz/build/util/hb-view googlesans-flex-fonts/fonts/variable/GoogleSansFlex\[ROND,opsz,slnt,wdth,wght\].ttf --glyphs divide --variations wdth=67.5 > out.png
```
![out](https://github.com/googlefonts/googlesans-flex/assets/142891/b11ed2f0-0b44-470d-ba85-57524804334d)

As you see, the dots shrank! That's because of the different contour start point between the two masters.

**Comment by belluzj** [comment]
Is this already or could this be a fontbakery check?

**Comment by behdad** [comment]
I believe it's a fontbakery check already.

**Comment by behdad** [comment]
> Glyph degree was not compatible: 
>     Contour order differs: [0, 1] in (), [1, 0] in (('opsz', 1.0), ('wght', -1.0))

I inspected this in samsa and can't see any issues. I'll debug the tool.

**Comment by MariannaPaszkowska** [comment]
Thanks for reporting that @behdad 

@belluzj I found the check proposal: https://github.com/fonttools/fontbakery/issues/3930 
And here is the documentation: https://font-bakery.readthedocs.io/en/latest/fontbakery/profiles/universal.html?highlight=interpolation%20issues#com.google.fonts/check/interpolation_issues

Are we including this check?

**Comment by behdad** [comment]
> > Glyph degree was not compatible:
> > Contour order differs: [0, 1] in (), [1, 0] in (('opsz', 1.0), ('wght', -1.0))
> 
> I inspected this in samsa and can't see any issues. I'll debug the tool.

Upon further investigation this one is also an issue:

At opsz=144,wght=1:
![out1](https://github.com/googlefonts/googlesans-flex/assets/142891/dc0d7062-0193-41b0-8632-dcd93eb06e02)

At opsz=131,wght=1:
![out0](https://github.com/googlefonts/googlesans-flex/assets/142891/835fb673-f2c0-4943-bd2c-3fdb300f4dd9)

**Comment by belluzj** [comment]
Thanks for the link Marianna! I'll make sure we include it in the config

**Comment by behdad** [comment]
With the acutecomb glyph I have not been able to fully reproduce the issue yet. But this just looks wrong to me from a weight perspective:

At ROND=100,opsz=144,wdth=100,wght=1:
![out1](https://github.com/googlefonts/googlesans-flex/assets/142891/6f754ed7-85a4-48ed-aca7-d682988c0bfe)

At ROND=100,opsz=144,wdth=151,wght=1:
![out](https://github.com/googlefonts/googlesans-flex/assets/142891/1fff0c96-0dd5-4135-8474-99f46573b64c)

**Comment by Hoolean** [comment]
@MariannaPaszkowska

Hi Marianna!

I have confirmed that we run the `interpolation_issues` check as part of the the Universal profile, with it [outputting what appear on the surface to be similar warnings](https://github.com/googlefonts/googlesans-flex/actions/runs/6457535291/job/17529262259#step:8:1460) for our most recent build.

There are a few issues with how the CI currently presents the output:

1. **Configuration: Check output is truncated, and so not all of the interpolation warnings are shown**
Should we configure FontBakery to show all of them, either just locally with `make build` or on the CI too?
    
2. **Bug: Warning locations are not serialised (e.g. `<fontTools.ttLib.ttGlyphSet._TTGlyphSetGlyf object at   
           0x7f3ca891aa50`)**
This was fixed in a recent FontBakery PR by Simon (fonttools/fontbakery#4288). We can move to a newer FontBakery version to receive this fix, although discussion on #282 suggests that this may require a wider dependency update, or a modification of our CI job to run FontBakery in its own independent venv. How would we like to prioritise this?

**Comment by MariannaPaszkowska** [comment]
@Hoolean I think it's a good moment to update all dependencies we need.

**Comment by MariannaPaszkowska** [comment]
It's ok to leave the output truncated and run with full-lists locally.

**Comment by Hoolean** [comment]
@MariannaPaszkowska

I'm going to look into this now; would we like to update **every** dependency, to go into V2 with the latest, or just the minimal set needed for the FontBakery update?

**Comment by Hoolean** [comment]
Update from earlier discussion: these checks will be accessible from the CI pending two changes:

- [x] [Makefile QA task](https://github.com/googlefonts/googlesans-flex/blob/66b886d4/Makefile#L60-L65) is adjusted to include GRAD in the largest VF's name (#751)
- [x] fonttools/fontbakery#4357 is released

Next steps:

- [x] Check that the check output matches the PDF

**Comment by behdad** [comment]
Here's a full report for the latest released version. Note that this version of the report might have many false positives and minor kinks / underweight issues. The more important issues are sorted first.

I also see a lot of "wrong_start_point" false-positives. I'll be looking into those.

Thanks,
[out.pdf](https://github.com/googlefonts/googlesans-flex/files/13642823/out.pdf)

**Comment by Hoolean** [comment]
@MariannaPaszkowska 

Hello Marianna! The FontBakery check for interpolation testing is now fixed and working; we just ran our FontBakery profile with the TTF from #753 and nearly all of the same glyphs are flagged as in the PDF attached there 🎉

There is one remaining exception: the FontBakery check only flags "contour order" and "wrong start point" issues, and so some of Behdad's recent improvements (e.g. "kink", "underweight", "overweight" warnings) are not flagged; would it be helpful for FontBakery to flag these also?
